qst/src/main/java/io/deephaven/qst/table/JoinBase.java

```diff
@@ -0,0 +1,16 @@
+package io.deephaven.qst.table;
+
+import io.deephaven.api.JoinAddition;
+import org.immutables.value.Value.Check;
+
+public abstract class JoinBase extends TableBase implements Join {
+
+    @Check
+    final void checkAdditions() {
+        if (additions().stream().map(JoinAddition::newColumn).distinct().count() != additions()
+            .size()) {
+            throw new IllegalArgumentException(
+                "Invalid join additions, must not use the same output column multiple times.");
```

Comment:
You were only "pretty sure" this is a real constraint.

Reply:
I could have sworn it was a real constraint, but only maybe it was on a different join, or only when only a LHS and not a RHS is specified. Is it bad to apply this constraint?

Reply:
I think this is a fine constraint for `join`. We don't allow full `SelectColumn`, just `MatchPair`, at which point there's really no value added by using the same LHS twice.